src/qml/qmltablefromlistmodelcolumn.cpp

```diff
@@ -0,0 +1,208 @@
+#include "qml/qmltablefromlistmodelcolumn.h"
+
+#include <QJSValue>
+#include <QString>
+#include <QtDebug>
+
+namespace {
+
+const QString displayRoleName = QStringLiteral("display");
+const QString decorationRoleName = QStringLiteral("decoration");
+const QString editRoleName = QStringLiteral("edit");
+const QString toolTipRoleName = QStringLiteral("toolTip");
+const QString statusTipRoleName = QStringLiteral("statusTip");
+const QString whatsThisRoleName = QStringLiteral("whatsThis");
+const QString fontRoleName = QStringLiteral("font");
+const QString textAlignmentRoleName = QStringLiteral("textAlignment");
+const QString backgroundRoleName = QStringLiteral("background");
+const QString foregroundRoleName = QStringLiteral("foreground");
+const QString checkStateRoleName = QStringLiteral("checkState");
+const QString accessibleTextRoleName = QStringLiteral("accessibleText");
+const QString accessibleDescriptionRoleName = QStringLiteral("accessibleDescription");
+const QString sizeHintRoleName = QStringLiteral("sizeHint");
+
+} // namespace
+
+namespace mixxx {
+namespace qml {
+QmlTableFromListModelColumn::QmlTableFromListModelColumn(QObject* parent)
+        : QObject(parent) {
+}
+QmlTableFromListModelColumn::~QmlTableFromListModelColumn() {
+}
+#define DEFINE_ROLE_PROPERTIES(getterGetterName,                              \
+        getterSetterName,                                                     \
+        getterSignal,                                                         \
+        setterGetterName,                                                     \
+        setterSetterName,                                                     \
+        setterSignal,                                                         \
+        roleName)                                                             \
+    QJSValue QmlTableFromListModelColumn::getterGetterName() const {          \
+        return mGetters.value(roleName);                                      \
+    }                                                                         \
+                                                                              \
+    void QmlTableFromListModelColumn::getterSetterName(                       \
+            const QJSValue& stringOrFunction) {                               \
+        if (!stringOrFunction.isString() && !stringOrFunction.isCallable()) { \
```

Comment:
Due to the macro nature all this code is duplicated. It would be better to use helper functions for this. 
This also helps to get around of some macro weirdness when debugging. 

Maybe there is a chance to omit using the macro entirely. Currently it has already 7 parameter. 
Maybe we can replace it wit 7 lines of c++ code forwarding to a template or something. 

If you wish to go with the macro solution you may consider to use only the property name as parameter.
Like shown here: https://github.com/mixxxdj/mixxx/blob/d67b05771269f2da309912a756d359e1a5ad260d/src/util/macros.h#L21 

But I would prefer a c++ based solution even if it involves more writing initaly.

Reply:
This macro-based solution works fine IMHO. If you want to replace the macros in a follow-up PR, feel free to do so.